kore/app/profiler/Main.hs

```diff
@@ -72,59 +110,119 @@ buildReport = buildReportHelper 0 initialContext Map.empty
                 case parseLine line of
                     Just parsed ->
                         let fixedParsed = fixTags parsed
-                            newContext = updateContext context fixedParsed
+                            newContext =
+                                updateContext profileOptions context fixedParsed
                         in buildReportHelper
                             (lineIndex + 1)
                             newContext
-                            $! addProfileEvent newContext fixedParsed report
+                            $! addProfileEventFiltered
+                                context newContext fixedParsed report
                     Nothing ->
                         buildReportHelper (lineIndex + 1) context report
 
 reportToText :: Map String EventSummary -> Text
 reportToText = writeReport . removeMatchingStarts
 
-updateContext :: Context -> ProfileEvent -> Context
+updateContext :: ProfileOptions -> Context -> ProfileEvent -> Context
 updateContext
+    profileOptions
     context
     ProfileEvent { tags, endPico = Nothing }
   =
-    foldr openTag context tags
+    withTags (withEnabled context)
   where
+    withEnabled :: Context -> Context
+    withEnabled Context { openedTagsToMultiplicity, enabled } =
```

Comment:
If `openedTagsToMultiplicity` is just passing through, you could use record update syntax:
```.hs
    withEnabled context@Context { enabled } =
        context { enabled = isEnabled enable : enabled }
```